client/components/main/index.jsx

```diff
@@ -17,3 +29,44 @@ module.exports = React.createClass( {
 		);
 	}
 } );
+
+/**
+ * As apparent below, the idea is simple: connect Main to Redux so it knows
+ * whether it should render a guided tour rather than `primary`.
+ *
```

Comment:
Right, I should've made the motivation clear:

The preview is currently overlaid, but if you try resizing you'll see a lot of funkiness going on, as the sidebar shrinks, then hides. Specifically, you'll see a glimpse of `primary` in between the sidebar and the overlay.

My reasoning, which I should've shared yesterday, is that it's probably a bad idea to rely on a `position: fixed` element to make sure we fully hide Calypso consistently across different viewports and wild scenarios. Thus, I'd like Main — since it hierarchically sits pretty close to `primary` (though sadly not _above_ it; suggestions for other components that would work across sections are welcome)  — to do the following, roughly speaking:

```js
const classes = classnames( this.props.className, 'main', {
    // I'm terrible at CSS-oriented naming
    'main__is-guidestour-preview-active': shouldShowGuides
} );
return (
  <main className={ classes } role="main">
    { shouldShowGuides && <WebPreview { ...previewProps } /> }
    { this.props.children }
  </main>
);
```

That would:

1. Insert the preview at the same document level of the content / `primary`, thus gaining more flexibility for good overlay positioning.
1. Mark `<main>` with a stateful class so as to make it easier to use plain CSS to hide a Calypso section while the preview is showing.

Is this sound? :)